lib/index.js

```diff
@@ -81,13 +81,21 @@ function createStringifyStream(opts) {
  * stream based JSON.parse. async function signature to abstract over streams.
  * @public
  * @param {Object} opts options to pass to parse stream
- * @param {String} opts.body string to parse
+ * @param {String|Buffer} opts.body string or buffer to parse
  * @param {Function} callback a callback function
  * @return {Object} the parsed JSON object
  */
 function _parse(opts, callback) {
     assert.object(opts, 'opts');
-    assert.string(opts.body, 'opts.body');
+    try {
+        assert.string(opts.body, 'opts.body');
+    } catch (e) {
+        if (e instanceof assert.AssertionError) {
+            assert.buffer(opts.body, 'opts.body');
+        } else {
+            throw e;
+        }
+    }
```

Comment:
Line 96 (`throw e`) is hard to test - we would have to mock `assert.string` which is why the coverall test breaks. If you want to merge this, we could remove the `if (e instanceof assert.AssertionError) {` or we take the coverall hit or you know another way to assert for either-or ?

Reply:
How about something like this?

`assert.ok(typeof opts.body === 'string' || Buffer.isBuffer(opts.body))` ?